### ISSUE TITLE
Type created with union loses its part after Omit

### ISSUE THREAD

**Issue by everdimension** [microsoft/TypeScript#43883]
# Bug Report

### 🔎 Search Terms

_omit loses type info union generic_

### 🕗 Version & Regression Information

Version: 4.2.4 

- I was unable to test this on prior versions because _omg it's exhausting_

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgEJwM4QDwBUB8yA3gLABQyyAHgFzK4A05lAnnSAK4C2ARtANzkAvuXJgWABxQBBACazgYYAHsQcADbIAvMWRx23PlGRDkAH1086GMFFABzE4LLipyAMpSEwGMAR5CHXQsAOQAMmQ5BSVVDWcxSRQAMVANUJ0AeS5FbE8Ib19-AgZkAHIWUvxRMgRVG2o6FLV1bE5eaEDdWmQAJgBmJ2QAeiHkLg56gHc4KBA9HmUOMDHgDAwHMrhS8zKeUqA)

### 💻 Code

```ts
interface Base<T> {
  x: T,
  y: number;
}

type Additional = { a: number } | { b: string };
type Specific<T> = Base<T> & Additional;

type Final<T> = Omit<Specific<T>, 'y'>

const x: Final<number> = { x: 23 }; // must warn about missing 'a' | 'b', but doesn't
```

### 🙁 Actual behavior

`Final` type doesn't have properties of type `Additional`

### 🙂 Expected behavior

`Final` type must properties of type `Additional`

**Comment by MartinJohns** [comment]
Duplicate of #39556.

**Comment by RyanCavanaugh** [comment]
>  omg it's exhausting

No one is forcing you to fill out this report; please just refrain from doing so if you aren't able to complete the steps asked of you. We put those steps in there for a reason and I assure you someone else will log the issue if it's legitimate.

**Comment by everdimension** [comment]
@RyanCavanaugh this wasn't meant as an attack. Just the first thing I felt when reading the template. Also the template comment asked to keep one line.
Thanks for understanding! 🙂
